joblib/_parallel_backends.py

```diff
@@ -286,7 +293,13 @@ def compute_batch_size(self):
             # while a couple of CPUs a left processing a few long running
             # batches. Better reduce the batch size a bit to limit the
             # likelihood of scheduling such stragglers.
-            batch_size = old_batch_size // 2
+
+            # decrease the batch size quickly to limit potential starving
+            ideal_batch_size = int(
+                old_batch_size * self.MIN_IDEAL_BATCH_DURATION / batch_duration
+            )
```

Comment:
Could be computed outside the `if` to avoid having twice the same expression (and avoid changing only one).

Reply:
Actually we cannot because `batch_duration` can be 0 outside of these clauses.

Reply:
I mean we can do a try/except but the added complexity will defeat the initial purpose...

Reply:
Can it really be 0? It is just a nitpick so don't complexify it just for this.

Another solution would be to send back `old_batch_size` if `batch_duration == 0` at the beginning but it is not so important so no worries.

Reply:
> Can it really be 0?

It can: I figured it out while testing what you suggested :smile: `batch_duration` is set to 0 at the beginning.

Reply:
Aha ok! then lets keep it simple.